src/utils/nextEntityId.ts

```diff
@@ -0,0 +1,15 @@
+import { EntityManager } from 'typeorm'
+import { NextEntityId } from '../model'
+
+// used to retrieve the next id for an entity
+export async function getNextIdForEntity(em: EntityManager, entityName: string): Promise<number> {
+  const id = parseInt(
+    (
+      await em.getRepository(NextEntityId).findOne({
+        where: { entityName },
+        lock: { mode: 'pessimistic_write' },
+      })
+    )?.nextId.toString() || '1'
+  )
+  return id
+}
```

Comment:
"The data types smallserial, serial and bigserial are not true types, but merely a notational convenience for creating unique identifier columns"
So I don't think it will fit the case, as there will be just one row for the `NextEntityId`

Reply:
Plus it would be complex to generate `@ PrimaryGeneratedColumn` for every column of the table, since it is not possible to do so with squid typegen